tests/test_extractors.py

```diff
@@ -91,3 +91,15 @@ def test_extract_from_directory(self, extractor, tmp_path, file):
     def test_extract_from_nonsense(self, extractor):
         with pytest.raises(NotImplementedError):
             extractor.extract(8)
+
+    @pytest.mark.parametrize(
+        "source",
+        [
+            """
+                def foo():
+            """,
+        ],
+    )
+    def test_extract_with_syntax_error(self, extractor, file_path, file, caplog):
```

Comment:
**suggestion (testing):** Add a test case for successful file extraction following a syntax error.

After testing the handling of files with syntax errors, it would be valuable to include a test case that verifies the extractor's ability to successfully process subsequent files without syntax errors. This ensures that a syntax error in one file does not halt or affect the processing of other valid files.